### ISSUE TITLE
Where is the primers_to_amplicons.py script?

### ISSUE THREAD

**Issue by juliadouglasf** [jts/ncov-tools#105]
Hello,

I can't find the `primers_to_amplicons.py` script within the [ncov-tools scripts directory](https://github.com/jts/ncov-tools/tree/master/workflow/scripts).

I am trying to run the [NML fork of the SIGNAL pipeline](https://github.com/phac-nml/covid-19-signal-nml), but everything seems to be running smoothly until the ncov-tools `make_amplicon_bed` rule. There is no obvious error; only `Error in rule make_amplicon_bed` ... `(one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)`. Hoping to solve the issue by looking at the script mentioned in the shell command, `primers_to_amplicons.py`, I realized it was missing.

Is the script still in use, as the Snakefile suggests, and if so, where might I find it?

Thanks.

**Comment by rdeborja** [comment]
It can be found in the `ncov-parser` package which is one of the dependencies in the `environment.yaml` file.

https://github.com/simpsonlab/ncov-parser/blob/main/bin/primers_to_amplicons.py

**Comment by DarianHole** [comment]
The conda env for that fork should be found in your clone of the repo under `.snakemake/conda/ncov-qc-pangolin4`. If ncov-parser is not installed there that would be odd as as Richard said its in the environment.yaml file. Hopefully doublechecking the install fixes it

It could be an issue with one of the primer bed files as well though

**Comment by juliadouglasf** [comment]
Thank you both for your responses! ncov-parser wasn't listed with `pip list` within the conda environment @DarianHole mentioned, so I manually used `pip install ncov-parser`. After `chmod +x primers_to_amplicons.py` to get rid of the ensuing `Permission denied` error, my colleague and I are troubleshooting a `ModuleNotFoundError` for `vcf`.

```
File ".../covid-19-signal-nml/.snakemake/conda/ncov-qc-pangolin4/bin/primers_to_amplicons.py", line 8, in <module>
    import ncov.parser.primers as pr
File ".../covid-19-signal-nml/.snakemake/conda/ncov-qc-pangolin4/lib/python3.8/site-packages/ncov/parser/__init__.py", line 7, in <module>
    from .Vcf import *
File ".../covid-19-signal-nml/.snakemake/conda/ncov-qc-pangolin4/lib/python3.8/site-packages/ncov/parser/Vcf.py", line 7, in <module>
    import vcf
ModuleNotFoundError: No module named 'vcf'
```

**Comment by DarianHole** [comment]
Did potentially none of the pip installs work? That is what it somewhat sounds like which would be odd. Actually it looks like I had a spacing difference in that file so that is likely it. Sorry about that. I've adjusted it

You'll probably either have to pip install the remaining dependencies or re-make the env after pulling the fix I made to the yaml file.

https://github.com/jts/ncov-tools/blob/master/workflow/envs/environment.yml#L31 has all the dependencies
